packages/core/svelte/src/walletStore.ts

```diff
@@ -84,13 +83,54 @@ export const walletStore = writable<WalletStore>({
     signMessage: undefined,
 });
 
-export const walletNameStore = writable<WalletNameStore>({
-    walletName: null,
-});
+function createWalletNameStore() {
+    const { subscribe, set } = writable<WalletNameStore>({
+        walletName: null
+    });
 
-export const walletAdapterStore = writable<WalletAdapterStore>({
-    adapter: null,
-});
+    function updateWalletName(walletName: WalletName | null) {
+        const { localStorageKey } = get(walletConfigStore);
+
+        set({ walletName });
+        setLocalStorage(localStorageKey, walletName);
+    }
+
+    return {
+        subscribe,
+        updateName: (walletName: WalletName | null) => updateWalletName(walletName),
```

Comment:
With the new semantics, don't think you should call `updateName` with `null`

Reply:
Not sure I understand this change, or the one below with `createWalletAdapterStore`. Why have an internal named setup function that is only called once? And why have `updateName` and `reset`? It seems like you could just do this if you're looking to decorate the `set` function and keep the API of the store.

```ts
export const walletNameStore = (function () {
    const { subscribe, set } = writable<WalletNameStore>({
        walletName: null
    });
    
    return {
        subscribe,
        set (walletName: WalletName | null) {
            const { localStorageKey } = get(walletConfigStore);

            set({ walletName });
            setLocalStorage(localStorageKey, walletName);
        }
    };
})();
```

Is it idiomatic to wrap stores like this? Or is there a way to react to changes to them?

Reply:
Your solution works, but it hides the intent. I wanted to make it clear that you're resetting the store, as when you're just reading the code, its not immediately clear that there's any side effects. It's not until I traced the code, I realised that updating the name to `null` triggers some side effects.

Its fairly common to create a custom store (wrap stores) to create a more clearly defined API, rather than just `set/update` and also a neat way to run side effect without having to listen to changes with a `.subscribe`

Reply:
There's an example of this in the [svelte tutorial](https://svelte.dev/tutorial/custom-stores)

```ts
function createCount() {
	const { subscribe, set, update } = writable(0);

	return {
		subscribe,
		increment: () => update(n => n + 1),
		decrement: () => update(n => n - 1),
		reset: () => set(0)
	};
}
```

Reply:
Got it, thanks for clarifying.